### DISCUSSION TITLE
fullScreenSourceRect doesnt work on Android?

### DISCUSSION THREAD

**Discussion by LilithRainbows** [airsdk/Adobe-Runtime-Support#2422] [Q&A]
fullScreenSourceRect works on Windows but doesnt seems to work on Android, or maybe im not using it right?
Windows example (good):
![image](https://user-images.githubusercontent.com/44578053/212457110-f6e3ac95-0907-4ab9-8e67-7dbd937d1441.png)
![image](https://user-images.githubusercontent.com/44578053/212457116-8bb8455d-692d-4cbe-87d3-8002e89afa3a.png)
Android example (bad):
![Screenshot_20230114-021618](https://user-images.githubusercontent.com/44578053/212457198-6cc3ac83-d791-410f-a12c-522b2a1297bc.png)
![Screenshot_20230114-021623](https://user-images.githubusercontent.com/44578053/212457200-837c719a-b436-414c-8ee1-be7bbdc103c8.png)

Code example:

```
import flash.display.Sprite;
import flash.display.Stage;
import flash.display.StageDisplayState;
import flash.events.MouseEvent;
import flash.geom.Rectangle;

// create red square on stage, turn it into a button for going to full screen
var redSquare:Sprite = new Sprite();
redSquare.graphics.beginFill(0xFF0000);
redSquare.graphics.drawRect(0, 0, 300, 300);
redSquare.x = 50;
redSquare.y = 50;
redSquare.addEventListener(MouseEvent.CLICK, enterFullScreen);
redSquare.buttonMode = true;
addChild(redSquare);
 
function enterFullScreen(e:MouseEvent):void
{
    // we will go to full screen zoomed in on the red square
    var redSquare:Sprite = e.target as Sprite;
    var fullScreenRect:Rectangle = new Rectangle(redSquare.x, redSquare.y, redSquare.width, redSquare.height);
 
    // calculate aspect ratio of the red square
    var rectAspectRatio:Number = fullScreenRect.width / fullScreenRect.height;
 
    // calculate aspect ratio of the screen
    var screenAspectRatio:Number = stage.fullScreenWidth / stage.fullScreenHeight;
  
    // change the fullScreenRect so that it covers the entire screen, keeping it centered on the redSquare
    // try commenting out this section to see what happens if you do not fix the aspect ratio.
    if (rectAspectRatio > screenAspectRatio) {
         var newHeight:Number = fullScreenRect.width / screenAspectRatio;
         fullScreenRect.y -= ((newHeight - fullScreenRect.height) / 2);
        fullScreenRect.height = newHeight;
    } else if (rectAspectRatio < screenAspectRatio) {
        var newWidth:Number = fullScreenRect.height * screenAspectRatio;
        fullScreenRect.x -= ((newWidth - fullScreenRect.width) / 2);
        fullScreenRect.width = newWidth;
    }
 
    // go to full screen
    stage.fullScreenSourceRect = fullScreenRect;
    stage.displayState = StageDisplayState.FULL_SCREEN;
}
```

**Comment by LilithRainbows** [comment]
Update:
Apparently, according to a 2011 news article, fullScreenSourceRect function had been fixed in AIR version 2.7, but I guess after all these years it must have broken again.
![image](https://user-images.githubusercontent.com/44578053/212672076-8e3e909a-a01d-46ae-9a63-0454d0ec9869.png)
Source:
https://news.softpedia.com/news/Adobe-AIR-v2-7-for-Android-Available-for-Download-206263.shtml

**Comment by DarkAlchem** [comment]
Trace the dimensions on mobile. I think there is a bug and you'd have to do some workarounds.

**Comment by allucaard3** [comment]
Hey Lillith, ive seen your work however i really need your help and i am extremely desperate, how can i bypass a ip ban on habbo? I can use a vpn to get onto habbo but the launcher is completley messed up with the ip ban, everytime i log in it says 'youve been banned!' even though the account isnt actually banned, its just a way to prevent from logging onto the launcher classic version of habbo - please can you help me out? i can give you my discord or something to get in touch with you ( im willing to pay if required)  i have a macbook btw, so im not sure what to do. i know with window its easier to spoof the mac address or another way around it as people have told me.
